Examples/Io/Root/include/ActsExamples/Io/Root/RootNuclearInteractionParametersWriter.hpp

```diff
@@ -0,0 +1,80 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020-2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "ActsExamples/Framework/WriterT.hpp"
+#include "ActsExamples/Io/Root/detail/NuclearInteractionParametrisation.hpp"
+
+#include <mutex>
+#include <vector>
+
+namespace ActsExamples {
+
+struct ExtractedSimulationProcess;
+
+/// @brief This class takes fractions of recorded events that represent the
+/// effect of a nuclear interaction and produces histograms and parameters which
+/// can be used for a parametrisation based simultion of nuclear interaction.
+/// Since the parameters are based on the set of all provided events, during the
+/// event loop newly provided events are stored until the end of the run. Then
+/// all parts are calculated and written to file.
+class RootNuclearInteractionParametersWriter final
+    : public WriterT<std::vector<ExtractedSimulationProcess>> {
+ public:
+  struct Config {
+    /// Input collection to map measured hits to simulated hits.
+    std::string inputSimulationProcesses;
+    /// output directory.
+    std::string outputDir;
+    /// output filename.
+    std::string outputFilename = "parameters.root";
+    /// file access mode.
+    std::string fileMode = "RECREATE";
+
+    /// Number of bins used for the interaction probability distributions
+    unsigned int interactionProbabilityBins = 1e6;
+    /// Number of bins used for the momentum distributions
+    unsigned int momentumBins = 1e6;
+    /// Number of bins used for the invariant mass distributions
+    unsigned int invariantMassBins = 1e6;
+    /// The highest final state multiplicity that will considered
+    unsigned int multiplicityMax = 10;
+    /// Choice whether the histograms should be written to file
+    bool writeHistograms = true;
```

Comment:
This refers to all histograms or to additional analysis histograms that are not required for the parameter extraction? If it is the latter, the name should probably be something like `writeOptionalHistograms`.

Reply:
changed